pkg/virtual/workspaces/registry/rest.go

```diff
@@ -441,12 +441,12 @@ func (w withProjection) ResultChan() <-chan watch.Event {
 			}
 			w.ch <- ev
 		}
+		defer close(w.ch)
```

Comment:
```suggestion
		close(w.ch)
```

Reply:
oh, this must go to the top of the function, i.e. above the for loop

Reply:
Sure.

Reply:
but I don't understand it, when the gorutine exits?
doesn't the `w.delegate.Stop()` close the `ch` channel?

Reply:
`ch` is being closed by `w.delegate.Stop()` but the program pointer in the goroutine might be at line 434 `w.ch <- ev`, sending data to a closed channel.

Reply:
fixed @sttts's comment